File: algorithms/ar-csrt/Python3/Count Sort.py
#Program for Count Sort by Animesh Tiwari
def countingSort(inputArray):
    maxElement= max(inputArray)
    countArrayLength = maxElement+1
    countArray = [0] * countArrayLength
    for el in inputArray: 
        countArray[el] += 1
        
    for i in range(1, countArrayLength):
        countArray[i] += countArray[i-1] 
    outputArray = [0] * len(inputArray)
    i = len(inputArray) - 1
    while i >= 0:
        currentEl = inputArray[i]
        countArray[currentEl] -= 1
        newPosition = countArray[currentEl]
        outputArray[newPosition] = currentEl
        i -= 1

    return outputArray

inputArray = [4,2,8,0,1,6,9,7]
print("Input array = ", inputArray)

sortedArray = countingSort(inputArray)
print("Counting sort result = ", sortedArray)

File: algorithms/ar-csrt/Javascript/count_sort.js
let countingSort = (arr, min, max) => {
  let i = min,
    j = 0,
    count = [];
  for (i; i <= max; i++) {
    count[i] = 0;
  }
  for (i = 0; i < arr.length; i++) {
    count[arr[i]] += 1;
  }
  for (i = min; i <= max; i++) {
    while (count[i] > 0) {
      arr[j] = i;
      j++;
      count[i]--;
    }
  }
  return arr;
};

let data = [4, 2, 2, 8, 3, 3, 1];
data = countingSort(data, 1, 8);
console.log(data);
